### PR TITLE
Relax deepdiff dependency.

### PR DESCRIPTION
**Pull request by Kircheneer**

### CODE REVIEW

**Comment by Kircheneer** [comment]
See https://github.com/nautobot/nautobot-plugin-golden-config/pull/498

**Comment by chadell** [comment]
this allows moving to a new major version (`6.0.0`). Not sure if this is desirable...

**Comment by Kircheneer** [comment]
Doesn't look problematic to me:

> 
> v6-0-0
> 
>     [Exclude obj callback strict](https://github.com/seperman/deepdiff/pull/320/files) parameter is added to DeepDiff by Mikhail Khviyuzov [mskhviyu](https://github.com/mskhviyu).
> 
>     A fix for diffing using iterable_compare_func with nested objects by [dtorres-sf](https://github.com/dtorres-sf) who originally contributed this feature.
> 

One looks like a bug fix and the other is a new feature.

https://zepworks.com/deepdiff/current/changelog.html

File: pyproject.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ include = [
 
 [tool.poetry.dependencies]
 python = "^3.7.0"
-deepdiff = "^5.5.0"
+deepdiff = ">=5.5.0"
```
Comment (pszulczewski) [inline pyproject.toml:26]:
```suggestion
deepdiff = ">=5.5.0 <7.0"
```
And we should be fine. I've checked all unittests are passing with 6.3.

**Review by pszulczewski — DISMISSED** [review]

**Comment by pszulczewski** [comment]
@Kircheneer you need to update poetry.lock

**Comment by Kircheneer** [comment]
I don't think so @pszulczewski - I am just allowing newer versions, I am not necessitating their use.

**Comment by pszulczewski** [comment]
From the CI:
``` Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.```

**Comment by lvrfrc87** [comment]
Would be nice to have it in poetry lock so CICD can run with the latest lib version an mack sure all test are ok

**Comment by Kircheneer** [comment]
Ahh I see.. done!

**Review by pszulczewski — APPROVED** [review]

**Pull request merged**